plugin/geoip/README.md

```diff
@@ -16,35 +18,54 @@ import (
 // ...
 if getLongitude := metadata.ValueFunc(ctx, "geoip/longitude"); getLongitude != nil {
     if longitude, err := strconv.ParseFloat(getLongitude(), 64); err == nil {
-		// Do something useful with longitude.
-	}
+        // Do something useful with longitude.
+    }
 } else {
     // The metadata label geoip/longitude for some reason, was not set.
 }
 // ...
 ```
 
 ## Databases
+
 The supported databases use city schema such as `City` and `Enterprise`. Other databases types with different schemas are not supported yet.
 
 You can download a [free and public City database](https://dev.maxmind.com/geoip/geolite2-free-geolocation-data).
 
 ## Syntax
-```txt
+
+```text
 geoip [DBFILE]
 ```
+
+or
+
+```text
+geoip [DBFILE] {
+    edns-subnet-processing
+}
+```
+
+By default, geoip plugin doesn't honor EDNS0 subnet extension, but it can be enabled using the extended syntax:
+
 * **DBFILE** the mmdb database file path.
+* `edns-subnet-processing` (optional): by default, geoip plugin read client IP and ignores [EDNS0 subnet](https://en.wikipedia.org/wiki/EDNS_Client_Subnet) extension. Using this option enables reading source IP from EDNS0 subnet when available. This help identifying closest source IP address as well as making GeoIP test processing easy: `dig +subnet=1.2.3.4 @dns-server.example.com www.geo-aware.com`.
 
 ## Examples
+
 The following configuration configures the `City` database.
+
 ```txt
 . {
-    geoip /opt/geoip2/db/GeoLite2-City.mmdb
+    geoip /opt/geoip2/db/GeoLite2-City.mmdb {
+      edns-subnet-processing
```

Comment:
The description of the example should reflect this. e.g. "The following configuration configures the `City` database, and looks up geo location based on edns0 subnet if present.

Reply:
Good point, thanks.

Reply:
I was reusing this conversation for edns-subnet processing benefits. Now a possible benefit for intermediary dns recursors is mentioned at "edns-subnet" config option.